src/library/dao/directorydao.cpp

```diff
@@ -149,32 +150,35 @@ QList<RelocatedTrack> DirectoryDAO::relocateDirectory(
     // mysterious ways for example if a track in the oldFolder also has a zombie
     // track location in newFolder then the replace query will fail because the
     // location column becomes non-unique.
+    const QString oldFolderLocation = mixxx::FileInfo(oldFolder).location();
+    DEBUG_ASSERT(!oldFolderLocation.endsWith('/'));
+    const QString newFolderLocation = mixxx::FileInfo(newFolder).location();
+    DEBUG_ASSERT(!newFolderLocation.endsWith('/'));
+
     QSqlQuery query(m_database);
     query.prepare("UPDATE " % kTable % " SET " % kLocationColumn %
-            "=:newFolder WHERE " % kLocationColumn % " = :oldFolder");
-    query.bindValue(":newFolder", newFolder);
-    query.bindValue(":oldFolder", oldFolder);
+            "=:newFolderLocation WHERE " % kLocationColumn % " = :oldFolderLocation");
+    query.bindValue(":newFolderLocation", newFolderLocation);
+    query.bindValue(":oldFolderLocation", oldFolderLocation);
     if (!query.exec()) {
         LOG_FAILED_QUERY(query) << "could not relocate directory"
-                                << oldFolder << "to" << newFolder;
+                                << oldFolderLocation << "to" << newFolderLocation;
         return {};
     }
 
-    // on Windows the absolute path starts with the drive name
-    // we also need to check for that
-    QString startsWithOldFolder = SqlLikeWildcardEscaper::apply(
-                                          QDir(oldFolder).absolutePath() + "/", kSqlLikeMatchAll) +
+    // The trailing path separator is crucial!
```

Comment:
because ... ?

Reply:
For disambiguation, what else. Otherwise you might match files instead of directories with the same prefix.

/a/b.mp3
/a/b/c.mp3

The prefix "a/b" matches both.